packages/components/src/TooltipTrigger.test.ts

```diff
@@ -0,0 +1,178 @@
+  ) => {
+    return renderWithTheme(
+      h(
+        TooltipTrigger,
+        {
+          content: tooltipContent,
+          ...props,
+        },
+        [button(childProps)]
+      )
+    );
+  };
+
+  it('shows tooltip when hovering on child', async () => {
+    // Arrange
+    const user = userEvent.setup();
+
+    renderTooltipTrigger();
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    await user.hover(btn);
+    const isShownAfterHover = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    await user.unhover(btn);
+    const isShownAfterUnhover = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    // Assert
+    expect(isShownAfterHover).toBe(true);
+    expect(isShownAfterUnhover).toBe(false);
+  });
+
+  it('shows tooltip when focusing on child', () => {
+    // Arrange
+    renderTooltipTrigger();
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    act(() => {
+      btn.focus();
+    });
+    const isShownAfterFocus = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    act(() => {
+      btn.blur();
+    });
+    const isShownAfterBlur = !!within(getPopupRoot()).queryByText(tooltipContent);
+
+    // Assert
+    expect(isShownAfterFocus).toBe(true);
+    expect(isShownAfterBlur).toBe(false);
+  });
+
+  it('calls child event handlers when hovering/unhovering', async () => {
+    // Arrange
+    const user = userEvent.setup();
+
+    const onMouseEnter = jest.fn();
+    const onMouseLeave = jest.fn();
+    renderTooltipTrigger({}, { onMouseEnter, onMouseLeave });
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    await user.hover(btn);
+    await user.unhover(btn);
+
+    // Assert
+    expect(onMouseEnter).toHaveBeenCalled();
+    expect(onMouseLeave).toHaveBeenCalled();
+  });
+
+  it('calls child event handlers when focusing/blurring', () => {
+    // Arrange
+    const onFocus = jest.fn();
+    const onBlur = jest.fn();
+    renderTooltipTrigger({}, { onFocus, onBlur });
+
+    const btn = screen.getByRole('button');
+
+    // Act
+    act(() => {
+      btn.focus();
+    });
+
+    act(() => {
+      btn.blur();
+    });
+
+    // Assert
+    expect(onFocus).toHaveBeenCalled();
+    expect(onBlur).toHaveBeenCalled();
+  });
+
+  it(
+    'allows delaying showing the tooltip',
+    withFakeTimers(async () => {
+      // Arrange
+      renderTooltipTrigger({ delay: 1000 });
+
+      const btn = screen.getByRole('button');
+
+      // Act
+      act(() => {
+        btn.focus();
+      });
+
+      const tooltip = within(getPopupRoot()).getByText(tooltipContent);
+
+      const isShownAfterFocus = tooltip.style.display !== 'none';
+
+      act(() => {
+        jest.advanceTimersByTime(1000);
+      });
+
+      const isShownAfterDelay = tooltip.style.display !== 'none';
+
+      // Assert
+      expect(isShownAfterFocus).toBe(false);
+      expect(isShownAfterDelay).toBe(true);
+    })
+  );
+
+  describe('accessibility', () => {
```

Comment:
❤️

Reply:
I was pleasantly surprised to discover that `@testing-library/jest-dom` has a11y related matchers like `toHaveAccessibleName`.
https://github.com/testing-library/jest-dom?tab=readme-ov-file#custom-matchers